_pytest/hookspec.py

```diff
@@ -489,8 +489,16 @@ def pytest_report_teststatus(report):
     Stops at first non-None result, see :ref:`firstresult` """
 
 
-def pytest_terminal_summary(terminalreporter, exitstatus):
-    """ add additional section in terminal summary reporting.  """
+def pytest_terminal_summary(config, terminalreporter, exitstatus):
```

Comment:
this is a breaking change, any caller of the hook would need to add the extra parameter

this needs to be addressed in pluggy first

Reply:
This hook is called only by `terminal.py`, and we've added parameters to hooks in the past, it's not clear to me why this hook in particular cannot receive an extra argument at this point.

Reply:
in this particular hooks its probably less of an issue, however all hooks and the required arguments they take are currently part of the public api

for example `pytest_deselected` is in dire need of a added `reason` argument however its one of the hooks that are supposed to be called by 3rd parties, and they would simply break

Reply:
I see what you mean regarding `pytest_deselected`, but AFAIK this is the only hook where we expect to be called by 3rd parties, all other hooks are called by pytest itself.

So IMHO your concern is valid but not for most hooks.

Having said all that, if you really want I can remove the parameter, it was not really required by the hook implementations I touched.

Reply:
No worries, removed it. 👍